pep/app/pods/application/controller.ts

```diff
@@ -38,6 +40,9 @@ export default class Application extends Controller {
     @tracked viewedPeriod: ViewPeriod = ViewPeriod.PAST_WEEK;
     @tracked searchTerms: SearchTermValue[] = [];
 
+    @tracked rightSidebarWidgets = this.configuration.base.global.cards.right;
+    @tracked leftSidebarWidgets = this.configuration.base.global.cards.left;
```

Comment:
I'd prefer to do something like

```
@tracked readonly rightSidebarWidgets = this.configuration.base.global.cards.right;
@tracked readonly leftSidebarWidgets = this.configuration.base.global.cards.left;
```

Not only is it more compact, but it also looks cleaner too. It also still doesn't allow you to overwrite it

Reply:
I think you are misusing `@tracked` here though. `@tracked` is doing a bunch of stuff behind the scenes to tell ember/glimmer to observe changes to it and all that jazz, which is redundant and not needed here, so while the code is more compact, its actually making the app/resource usage heavier. Using getters for this is also already an established pattern throughout the app, so its not great that we'd have 2 patterns for the same thing now.

We can possibly achieve a similar compact/clean pattern like that tho by creating a custom "alias" decorator, which is really what you are after here, that would be alot more lightweight. We did something similar in some of our addons with an `@argDefault('...')` decorator, but basically it could look like this:

```ts
@alias('configuration.base.global.cards.left') readonly rightSidebarWidgets: Widgets[];
```
and internally, the decorator would basically do exactly what the usual getter pattern does.

Reply:
From what I remember, a getter is just auto tracked. So the only different would be manually telling it to be tracked?

Reply:
I dont think so, getters just auto-recalculate whenever one of the properties they use change which is just ES6 class functionality, but in ember-land, if the updated properties used by the gettter are not `@tracked`, the getter will still recalc, but it won't trigger a re-render, iirc.

By explicitly adding `@tracked` here, you are essentially creating a duplicate tracking observer for a property, `this.configuration.base`, that is already tracked.

Also, I think there might be another issue w/this pattern that I just realized. Here, you are just defaulting/initializing the `rightSidebarWidgets` field with the current value of `this.configuration.base.global.cards.right` on initial render. But if that configuration were to later change (e.g. if the user changes their language or somthing, we reload all configs from the API), it would not have the updated value. In this case, its probably unlikely an action like that would change these specific admin configs, but we should probably avoid that to be safe, and to set the pattern for other configs that may change.